CI: Fix botocore Error

alimcmaster1 opened the pull request:
- [x] closes #34086

AWS Service Exceptions are not statically defined in boto3. Easiest to just catch a ClientError here.

Ref Docs:
https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html#parsing-error-responses-and-catching-exceptions-from-aws-services

alimcmaster1 committed:
Update Error Msg

alimcmaster1 committed:
Fix botocore error

alimcmaster1 committed:
Add match pattern

pep8speaks commented:
Hello @alimcmaster1! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/tests/io/parser/test_network.py`](https://github.com/pandas-dev/pandas/blob/ae6139ced3b165f382ccc8db94baacdacc01cff2/pandas/tests/io/parser/test_network.py):

> [Line 179:89](https://github.com/pandas-dev/pandas/blob/ae6139ced3b165f382ccc8db94baacdacc01cff2/pandas/tests/io/parser/test_network.py#L179): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 88 characters)

alimcmaster1 committed:
black

jbrockmendel commented:
also need to catch in test_write_s3_parquet_fails

alimcmaster1 committed:
Add AWS Docs & Fix Parquet Test

alimcmaster1 committed:
Docs

alimcmaster1 committed:
Docs

alimcmaster1 commented:
https://github.com/pandas-dev/pandas/pull/34087/checks?check_run_id=658833815
Azure CI has run here^

Fixed up parquet test too and added boto docs on dynamic errors here

@jbrockmendel @jreback

jreback commented:
thanks @alimcmaster1  for the quick response
